software/NRG_itho_wifi/src/IthoSystem.cpp

```diff
@@ -258,29 +258,22 @@ void processSettingResult(const uint8_t index, const bool loop)
   root["loop"] = loop;
   if (resultPtr2410 != nullptr && ithoSettingsArray != nullptr)
   {
-    int64_t val0, val1, val2;
-    val0 = cast_raw_bytes_to_int(resultPtr2410 + 0, ithoSettingsArray[index].length, ithoSettingsArray[index].is_signed);
-    val1 = cast_raw_bytes_to_int(resultPtr2410 + 1, ithoSettingsArray[index].length, ithoSettingsArray[index].is_signed);
-    val2 = cast_raw_bytes_to_int(resultPtr2410 + 2, ithoSettingsArray[index].length, ithoSettingsArray[index].is_signed);
-
-    if (*(resultPtr2410 + 0) == 0x5555AAAA && *(resultPtr2410 + 1) == 0xAAAA5555 && *(resultPtr2410 + 2) == 0xFFFFFFFF) //something went wrong, indicate error, better handling needed
-    {
+		double cur = 0.0;
+		double min = 0.0;
+		double max = 0.0;
+
+		if (decodeQuery2410(resultPtr2410, &ithoSettingsArray[index], &cur, &min, &max))
+		{
+			root["Current"] = cur;
+			root["Minimum"] = min;
+			root["Maximum"] = max;
+		}
+		else
+		{
       root["Current"] = "error";
       root["Minimum"] = "error";
       root["Maximum"] = "error";
-    }
-    else if (ithoSettingsArray[index].type == ithoSettings::is_int)
-    {
-      root["Current"] = val0;
-      root["Minimum"] = val1;
-      root["Maximum"] = val2;
-    }
-    else
-    {
-      root["Current"] = static_cast<double>((int32_t) val0) / ithoSettingsArray[index].divider;
-      root["Minimum"] = static_cast<double>((int32_t) val1) / ithoSettingsArray[index].divider;
-      root["Maximum"] = static_cast<double>((int32_t) val2) / ithoSettingsArray[index].divider;
-    }
```

Comment:
I moved this to a function so we don't have to repeat this logic whenever we want to update settings (e.g. using MQTT in the future, if desired).